apps/backoffice-v2/src/domains/chat/chatbot-opengpt.tsx

```diff
@@ -1,38 +1,93 @@
-import { getClient, Webchat, WebchatProvider } from '@botpress/webchat';
+import { getClient, Webchat, WebchatProvider, WebchatClient } from '@botpress/webchat';
 import { buildTheme } from '@botpress/webchat-generator';
-import { useEffect } from 'react';
+import { useCallback, useEffect } from 'react';
 import { useAuthenticatedUserQuery } from '../../domains/auth/hooks/queries/useAuthenticatedUserQuery/useAuthenticatedUserQuery';
+import { useCurrentCaseQuery } from '../../pages/Entity/hooks/useCurrentCaseQuery/useCurrentCaseQuery';
+import { useLocation } from 'react-router-dom';
 
 // declare const themeNames: readonly ["prism", "galaxy", "dusk", "eggplant", "dawn", "midnight"];
 const { theme, style } = buildTheme({
   themeName: 'galaxy',
   themeColor: 'blue',
 });
 
-const clientId = '8f29c89d-ec0e-494d-b18d-6c3590b28be6';
-
 const Chatbot = ({
   isWebchatOpen,
   toggleIsWebchatOpen,
+  client,
+  setClient,
+  chatbotClientId,
 }: {
   isWebchatOpen: boolean;
   toggleIsWebchatOpen: () => void;
+  client: WebchatClient | null;
+  setClient: (client: WebchatClient) => void;
+  chatbotClientId: string;
 }) => {
-  const client = getClient({ clientId });
   const { data: session } = useAuthenticatedUserQuery();
+  const { data: currentCase } = useCurrentCaseQuery();
+  const { pathname } = useLocation();
```

Comment:
Change the use of `useLocation` and `pathname` with `const { entityId: workflowId } = useParams();`